sktime/forecasting/compose/_dmlf.py

```diff
@@ -0,0 +1,413 @@
+        self.exposure_vars = exposure_vars
+
+        super().__init__()
+
+        # Set default forecaster_res if not provided
+        if self.forecaster_res is None:
+            from sktime.forecasting.compose import make_reduction
+
+            self.forecaster_res = make_reduction(
+                LinearRegression(), strategy="recursive"
+            )
+
+        # Validate exposure_vars
+        if self.exposure_vars is None or len(self.exposure_vars) == 0:
+            raise ValueError(
+                "exposure_vars must be specified and contain at least one variable"
+            )
+
+        # Update tags based on component forecasters
+        self._update_tags_from_components()
+
+    def _update_tags_from_components(self):
+        """Update forecaster tags based on component capabilities."""
+        # Missing values capability - all components must support it
+        miss_y = self.forecaster_y.get_tag("capability:missing_values", False)
+        miss_ex = self.forecaster_ex.get_tag("capability:missing_values", False)
+        miss_res = self.forecaster_res.get_tag("capability:missing_values", False)
+
+        # Prediction interval capability - all components must support it
+        pred_int_y = self.forecaster_y.get_tag("capability:pred_int", False)
+        pred_int_ex = self.forecaster_ex.get_tag("capability:pred_int", False)
+        pred_int_res = self.forecaster_res.get_tag("capability:pred_int", False)
+
+        # In-sample prediction capability
+        insample_y = self.forecaster_y.get_tag("capability:insample", False)
+        insample_ex = self.forecaster_ex.get_tag("capability:insample", False)
+        insample_res = self.forecaster_res.get_tag("capability:insample", False)
+
+        self.set_tags(
+            **{
+                "capability:missing_values": miss_y and miss_ex and miss_res,
+                "capability:pred_int": pred_int_y and pred_int_ex and pred_int_res,
+                "capability:insample": insample_y or insample_ex or insample_res,
+            }
+        )
+
+    def _split_exogenous_data(self, X):
+        """Split exogenous data into exposure and confounder variables.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Exogenous variables
+
+        Returns
+        -------
+        X_ex : pd.DataFrame
+            Exposure variables
+        X_conf : pd.DataFrame
+            Confounder variables
+        """
+        if X is None:
+            raise ValueError("X cannot be None for DML forecasting")
+
+        # Check that all exposure variables exist in X
+        missing_vars = set(self.exposure_vars) - set(X.columns)
+        if missing_vars:
+            raise ValueError(
+                f"Exposure variables {missing_vars} not found in X columns: "
+                f"{list(X.columns)}"
+            )
+
+        X_ex = X[self.exposure_vars].copy()
+        X_conf = X.drop(columns=self.exposure_vars).copy()
```

Comment:
I think it should be explicitly mentioned in the documentation that non-exposure variables will be considered as confounders. Treating non-confounding variables as confounders can lead to biased or high-variance predictions.